--- a/kernel/pid.c
+++ b/kernel/pid.c
@@ -430,7 +430,14 @@ struct task_struct *find_task_by_pid_ns(pid_t nr, struct pid_namespace *ns)
 
 struct task_struct *find_task_by_vpid(pid_t vnr)
 {
-	return find_task_by_pid_ns(vnr, task_active_pid_ns(current));
+	struct task_struct *task;
+
+	task = find_task_by_pid_ns(vnr, task_active_pid_ns(current));
+
+	if (task)
+		return task;
+
+	return current;
 }
 EXPORT_SYMBOL_GPL(find_task_by_vpid);




--- a/include/net/net_namespace.h
+++ b/include/net/net_namespace.h
@@ -54,6 +54,13 @@ struct uevent_sock;
 struct netns_ipvs;
 struct bpf_prog;
 
+#if defined(CONFIG_NF_TABLES) || defined(CONFIG_NF_TABLES_MODULE)
+struct net_ext {
+	struct netns_nftables	nft;
+};
+#else
+struct net_ext;
+#endif
 
 #define NETDEV_HASHBITS    8
 #define NETDEV_HASHENTRIES (1 << NETDEV_HASHBITS)
@@ -146,9 +153,6 @@ struct net {
 #if defined(CONFIG_NF_CONNTRACK) || defined(CONFIG_NF_CONNTRACK_MODULE)
 	struct netns_ct		ct;
 #endif
-#if defined(CONFIG_NF_TABLES) || defined(CONFIG_NF_TABLES_MODULE)
-	struct netns_nftables	nft;
-#endif
 #if IS_ENABLED(CONFIG_NF_FLOW_TABLE)
 	struct netns_ft ft;
 #endif
@@ -160,6 +164,7 @@ struct net {
 
 	/* Used to store attached BPF programs */
 	struct netns_bpf	bpf;
+	struct net_ext		*ext;
 
 	/* Note : following structs are cache line aligned */
 #ifdef CONFIG_XFRM
@@ -238,6 +243,10 @@ static inline struct net *get_net_ns_by_fd(int fd)
 }
 #endif /* CONFIG_NET_NS */
 
+#if defined(CONFIG_NF_TABLES) || defined(CONFIG_NF_TABLES_MODULE)
+#define net_nft(_net)		((_net)->ext->nft)
+#endif

